src/store/addedSafesSlice.ts

```diff
@@ -49,10 +51,35 @@ export const addedSafesSlice = createSlice({
         delete state[chainId]
       }
     },
+    pinSafe: (state, { payload }: PayloadAction<{ chainId: string; address: string; removeOnUnpin: boolean }>) => {
+      const { chainId, address, removeOnUnpin } = payload
+      state[chainId] ??= {}
+      state[chainId][address] = {
+        ...(state[chainId][address] ?? {}),
+        pinned: true,
+        removeOnUnpin,
+      }
+    },
+    unpinSafe: (state, { payload }: PayloadAction<{ chainId: string; address: string }>) => {
+      const { chainId, address } = payload
+      const removeOnUnpin = state[chainId]?.[address]?.removeOnUnpin
+
+      if (removeOnUnpin) {
+        delete state[chainId]?.[address]
+        if (Object.keys(state[chainId]).length === 0) {
+          delete state[chainId]
+        }
+      } else {
+        state[chainId][address] = {
```

Comment:
This crashes for me sometimes when unpinning a multichain safe because `state[chainId]` is undefined.